python/test/unit/language/test_core.py

```diff
@@ -1981,6 +1981,70 @@ def kernel_dynamic(out, val, dtype: tl.constexpr):
 # test arange
 # ---------------
 
+def test_vecmat():
```

Comment:
Hmm this seems too coarse-grained for `test_core`. Could you move this to https://github.com/openai/triton/blob/main/python/test/regression/test_functional_regressions.py and maybe modify `test_convert2d` to support #shared layouts?

Reply:
Oh I was working on it already, will update soon.